internal/platform/web/errors.go

```diff
@@ -0,0 +1,28 @@
+package web
+
+import "net/http"
+
+type HTTPStatuser interface {
+	HTTPStatus() int
+}
+
+type statusError struct {
+	error
+	status int
+}
+
+func (e statusError) HTTPStatus() int {
```

Comment:
We need to use pointer semantics here for sure.

Reply:
Wait, Maybe not. My brain thought we were implementing the error interface for some reason.

Reply:
This is nothing more than a `getter`, so why do we need it. Exposing the `status` field will accomplish the same thing.

Reply:
This means we don't need the interface. I don't see a second implementation so honestly, why do we need the decoupling.

Reply:
Can't `statusError` implement the `error` interface and we can accomplish the same thing?

Reply:
I see your point on not needing the decoupling the interface provides, a concrete type probably makes sense here.